src/components/link.js

```diff
@@ -0,0 +1,345 @@
+      el.getObject3D('mesh').visible = true;
+      sphereEl.setAttribute('visible', false);
+    }
+  },
+
+  play: function () {
+    this.updateEventListener();
+  },
+
+  pause: function () {
+    this.removeEventListener();
+  },
+
+  updateEventListener: function () {
+    var el = this.el;
+    if (!el.isPlaying) { return; }
+    this.removeEventListener();
+    el.addEventListener(this.data.on, this.redirect);
+  },
+
+  removeEventListener: function () {
+    var on = this.data.on;
+    if (!on) { return; }
+    this.el.removeEventListener(on, this.redirect);
+  },
+
+  initVisualAspect: function () {
+    var el = this.el;
+    var textEl;
+    var sphereEl;
+    var semiSphereEl;
+    if (!this.data.visualAspectEnabled) { return; }
+    textEl = this.textEl = this.textEl || document.createElement('a-entity');
+    sphereEl = this.sphereEl = this.sphereEl || document.createElement('a-entity');
+    semiSphereEl = this.semiSphereEl = this.semiSphereEl || document.createElement('a-entity');
+
+    // Set Portal
+    el.setAttribute('geometry', {primitive: 'circle', radius: 1.5, segments: 64});
+    el.setAttribute('material', {
+      shader: 'portal',
+      pano: this.data.src,
+      side: 'double'
+    });
+
+    // Set text that displays the link title / url
+    textEl.setAttribute('text', {
+      color: 'white',
+      align: 'center',
+      font: 'kelsonsans',
+      value: this.data.title || this.data.href,
+      width: 4
+    });
+    textEl.setAttribute('position', '0 1.9 0');
+    el.appendChild(textEl);
+
+    // Set the sphere that is rendered when the camera is close
+    // to the portal to allow the user peek inside
+    semiSphereEl.setAttribute('geometry', {
+      primitive: 'sphere',
+      radius: 1.5,
+      phiStart: 0,
+      segmentsWidth: 64,
+      segmentsHeight: 64,
+      phiLength: 180,
+      thetaStart: 0,
+      thetaLength: 360
+    });
+
+    semiSphereEl.setAttribute('material', {
+      shader: 'portal',
+      borderEnabled: 0.0,
+      pano: this.data.src,
+      side: 'back'
+    });
+    semiSphereEl.setAttribute('rotation', '0 180 0');
+    semiSphereEl.setAttribute('position', '0 0 0');
+    semiSphereEl.setAttribute('visible', false);
+    el.appendChild(semiSphereEl);
+
+    // Set the sphere that is rendered when the camera is close
+    // to the portal to allow the user peek inside
+    sphereEl.setAttribute('geometry', {
+      primitive: 'sphere',
+      radius: 10,
+      segmentsWidth: 64,
+      segmentsHeight: 64
+    });
+
+    sphereEl.setAttribute('material', {
+      shader: 'portal',
+      borderEnabled: 0.0,
+      pano: this.data.src,
+      side: 'back'
+    });
+    sphereEl.setAttribute('visible', false);
+    el.appendChild(sphereEl);
+  },
+
+  redirect: function () {
+    window.location = this.data.href;
+  },
+
+  /**
+   * The tick handles:
+   * 1. It swaps the plane the represents the portal with a sphere with a hole when the camera is close
```

Comment:
Don't need to put `It` behind anywhere since it's ambiguous and we can infer from context